### PR TITLE
community: glob multiple patterns when using DirectoryLoader

### PR DESCRIPTION
**Pull request by finlaymacklon**
- **Description:** Updated *community.langchain_community.document_loaders.directory.py* to enable the use of multiple glob patterns in the `DirectoryLoader` class. Now, the glob parameter is of type `list[str] | str` and still defaults to the same value as before. I updated the docstring of the class to reflect this, and added a unit test to *community.tests.unit_tests.document_loaders.test_directory.py* named `test_directory_loader_glob_multiple`. This test also shows an example of how to use the new functionality.
- ~~Issue:~~**Discussion Thread:** https://github.com/langchain-ai/langchain/discussions/18559
- **Dependencies:** None
- **Twitter handle:** N/a

- [x] **Add tests and docs**
    - Added test (described above)
    - Updated class docstring

- [x] **Lint and test**

### CODE REVIEW

File: libs/community/tests/unit_tests/document_loaders/test_directory.py [hunk 1]
```diff
@@ -56,3 +57,35 @@ def test_exclude_ignores_matching_files(tmp_path: Path) -> None:
...
+
+    path_to_examples = "tests/examples/"
+    list_extensions = [".rst", ".txt"]
+    list_globs = [f"**/*{ext}" for ext in list_extensions]
+    is_file_type_loaded = {ext: False for ext in list_extensions} 
+
+    loader = DirectoryLoader(
+        path=path_to_examples,
+        glob=list_globs,
+        loader_cls=TextLoader
```
Comment (isahers1) [inline libs/community/tests/unit_tests/document_loaders/test_directory.py:73]: Can we use CustomLoader here so as to not change the imports? @baskaryan

File: libs/community/langchain_community/document_loaders/directory.py [hunk 2]
```diff
@@ -124,7 +124,14 @@ def lazy_load(self) -> Iterator[Document]:
         if not p.is_dir():
             raise ValueError(f"Expected directory, got file: '{self.path}'")
 
-        paths = p.rglob(self.glob) if self.recursive else p.glob(self.glob)
+        # glob multiple patterns if a list is provided, e.g., multiple file extensions
+        if type(self.glob) == list:
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/directory.py:128]: Maybe this?

```
if isinstance(self.glob, (list, tuple)):
  ...
elif isinstance(self.glob, str):
  ...
else:
  raise TypeError('Expected glob to be str or sequence of str, but got {type(self.glob)})
```

Comment (finlaymacklon) [reply libs/community/langchain_community/document_loaders/directory.py:128]: Sounds good! I will make that change on my fork soon. Thank you

File: libs/community/tests/unit_tests/document_loaders/test_directory.py (hunk 1 above)
Comment (finlaymacklon) [reply libs/community/tests/unit_tests/document_loaders/test_directory.py:73]: I will take a look at replacing TextLoader with CustomLoader briefly. Should be able to make the suggested change on my fork shortly. Thank you

**Comment by finlaymacklon** [comment]
Hello @isahers1 and @eyurtsev - I have now pushed changes that should address your comments on this pull request. Please let me know if any other changes are needed! Thank you

**Comment by anshumankmr** [comment]
Is there an ETA for when will this be released?

**Comment by anshumankmr** [comment]
Tested using langchain==0.2.11, it works for me.

**Pull request merged**